src/librustdoc/clean/mod.rs

```diff
@@ -3428,7 +3428,9 @@ fn build_deref_target_impls(cx: &DocContext,
         let primitive = match *target {
             ResolvedPath { did, .. } if did.is_local() => continue,
             ResolvedPath { did, .. } => {
-                ret.extend(inline::build_impls(cx, did, true));
+                // We set the last parameter to false to avoid looking for auto-impls for traits
+                // and therefore avoid an ICE.
```

Comment:
This is only part of the story. The real reason we should pass `false` here is because auto-traits don't propagate through `Deref` anyway, so we shouldn't be synthesizing impls for them.